src/pip/_internal/cache.py

```diff
@@ -168,11 +171,17 @@ def get(
         # type: (...) -> Link
         candidates = []
 
-        for wheel_name in self._get_candidates(link, package_name):
+        if not package_name:
+            return link
+
+        canonical_package_name = canonicalize_name(package_name)
+        for wheel_name in self._get_candidates(link, canonical_package_name):
             try:
                 wheel = Wheel(wheel_name)
             except InvalidWheelFilename:
                 continue
+            if wheel.name != canonical_package_name:
```

Comment:
To be safe, you should also canonicalize `wheel.name` like in:
https://github.com/pypa/pip/blob/4c1a7cda5a250b5b90824b720c71ec23fedbb943/src/pip/_internal/index/package_finder.py#L190

And maybe put a `logger.debug()` log for this rare case.

Reply:
Done